python/python/bystro/prs/tests/test_prs_dosage.py

```diff
@@ -0,0 +1,277 @@
+weights_with_missing = pd.concat(
+    [weights, pd.Series([0.1, 0.2, 0.3, 0.4, 0.5], index=["new1", "new2", "new3", "new4", "new5"])]
+)
+
+
+@pytest.fixture
+def create_feather_file(tmp_path):
+    # Use pytest's temporary directory feature
+    file_path = str(tmp_path / "test_dosage.feather")
+
+    # write an IPC/feather file
+    with pa.OSFile(file_path, "wb") as sink, pa.RecordBatchFileWriter(sink, table.schema) as writer:
+        writer.write_table(
+            table,
+        )
+
+    return file_path
+
+
+@pytest.fixture
+def create_feather_file_with_multiple_batches(tmp_path, batches=5):
+    # Use pytest's temporary directory feature
+    file_path = str(tmp_path / "test_dosage.feather")
+
+    # write an IPC/feather file
+    weights = {}
+    with pa.OSFile(file_path, "wb") as sink, pa.RecordBatchFileWriter(sink, table.schema) as writer:
+        for i in range(batches):
+            # generates a random set of 10 loci, taking chromosomes from chr1-22,
+            # positions from 1-100000000, and alleles from ACGT
+            loci = [
+                f"chr{random.randint(1,22)}:{random.randint(1,100000000)}:{random.choice(['A','C','G','T'])}:{random.choice(['A','C','G','T'])}"
+                for i in range(10)
+            ]
+            # generates a random set of 10 genotypes, taking values from 0-2, for 3 samples
+            data = {
+                "locus": loci,
+                "1805": np.random.randint(0, 3, size=len(loci), dtype=np.uint16),  # noqa: NPY002
+                "1847": np.random.randint(0, 3, size=len(loci), dtype=np.uint16),  # noqa: NPY002
+                "4805": np.random.randint(0, 3, size=len(loci), dtype=np.uint16),  # noqa: NPY002
+            }
+            print(data)
+            # write the batch of data to our feather file
+            writer.write_table(pa.Table.from_pandas(pd.DataFrame(data)))
+
+            weights.update({locus: random.random() for locus in loci})  # noqa: NPY002
+
+    return file_path, pd.Series(weights)
+
+
+def test_prs_calculation(create_feather_file):
+    """
+    Test PRS calculations with Pandas
+    This will be more memory efficient than going through Pandas
+    """
+    file_path = create_feather_file
+
+    # Read the Feather file into a PyArrow Table
+    with pa.memory_map(file_path, "r") as source:
+        table = pa.ipc.open_file(source).read_all()
+
+    # Compute the weighted sum for each sample column
+    prs_scores = {}
+    for column in table.column_names:
+        if column != "locus":
+            # Multiply the column by the weights and sum
+            weighted_column = pc.multiply(table.column(column), pa.array(weights))
+            prs_scores[column] = pc.sum(weighted_column).as_py()
+
+    assert len(prs_scores) == len(table.column_names) - 1  # excluding 'locus' column
+    print(prs_scores)
+
+
+def calculate_prs_in_pandas(file_path, weights):
+    """
+    Calculate PRS scores efficiently, in batches
```

Comment:
Since we decided we're doing PRS scores by sample rather than in batches, do we still need this test or do you want to leave it as an example?

Reply:
Follow up question, can you batch by samples rather than by loci?

Reply:
Yep! (you can grab [sample1, sample2, sample3] instead of [sample1])